ass2.cpp

```diff
@@ -127,23 +104,64 @@ void test_task1() {
 // of two nodes in a binary tree. Avoid storing additional nodes in a data structure.
 //
 
-struct node* find_low_com_ancestor(struct node* root, int n1, int n2) {
+struct node* find_low_com_ancestor(struct node* root, int n1, int n2, pair<bool, bool> &values_exists) {
     if (root == nullptr) {
         return nullptr;
     }
     
     if (root -> data == n1 || root -> data == n2) {
+        (root -> data == n1) ? values_exists.first = true : values_exists.second = true;
         return root;
     }
     
-    node* left_lca  = find_low_com_ancestor(root -> left_child, n1, n2);
-    node* right_lca = find_low_com_ancestor(root -> right_child, n1, n2);
+    node* left_lca  = find_low_com_ancestor(root -> left_child, n1, n2, values_exists);
+    node* right_lca = find_low_com_ancestor(root -> right_child, n1, n2, values_exists);
     
     if (left_lca && right_lca) {
         return root;
     }
+
+    return (left_lca != nullptr) ? left_lca : right_lca;
+}
+
+struct node* return_low_com_ancestor(struct node* root, int n1, int n2) {
+    pair<bool, bool> values_exists;
```

Comment:
This initialises to false, false already, if you want to make it explicit you can use:
pair<bool, bool> values_exists(false, false) ;
or
pair<bool, bool> values_exists{false, false};